backend/lib/endpoints/users.js

```diff
@@ -554,6 +565,143 @@ async function routes(app) {
       return updatedUser.notifyPrefs;
     },
   );
+
+  app.get(
+    "/verification",
+    { preValidation: [app.authenticate, app.setActor] },
+    async (req) => {
+      const { actor, userId } = req;
+      const [userErr, user] = await app.to(
+        BaseUser.findById(actor ? actor._id : userId),
+      );
+      if (userErr) {
+        req.log.error(userErr, "Failed retrieving user");
+        throw app.httpErrors.internalServerError();
+      } else if (user === null) {
+        req.log.error(userErr, "User does not exist");
+        throw app.httpErrors.notFound();
+      }
+      const sessionUrl = await Veriff.createSessionUrl(user);
+      return {
+        sessionUrl,
+      };
+    },
+  );
+
+  // Veriff decision webhook listener
+  app.post(
+    "/verification/decision",
+    { preValidation: [Veriff.validateWebhookEvent] },
+    async (req) => {
+      const {
+        body: { verification },
+      } = req;
+
+      if (verification && verification.vendorData) {
+        const {
+          id,
+          status,
+          vendorData: userId,
+          decisionTime,
+          reasonCode,
+        } = verification;
+
+        const [userErr, user] = await app.to(BaseUser.findById(userId));
+        if (userErr || user === null) {
+          // ALWAYS return true (200 code) when responding to the webhook event
```

Comment:
what if DB failed to read/write, will there be retry or other way to make sure the document get updated?

Reply:
We can trigger the webhook event from Veriff's dashboard if it failed.
you can check Papertrail logs for verified-badges, it was spamming the console very badly before this, because of some fake user IDs I used for testing, the same thing will happen in PROD after we add account deletion (knowing that Veriff will post a webhook event for that unexisiting user id later when their verification expires, and it will keep sending it every 2 minutes as long as we are not returning 200)
![image](https://user-images.githubusercontent.com/46847212/102633785-30187e00-4151-11eb-84f5-3a369e9ae5c6.png)

But I think it would be good to at least log that something went wrong, so we can trigger the webhook from Veriff again for that id.